### PR TITLE
[stdlib] Implement Never conformance to Codable

### PR DESCRIPTION
**Pull request by natecook1000**
This supplies conditional conformance for `Never` to the `Codable` protocol. 

Pitch thread: https://forums.swift.org/t/pitch-conform-never-to-codable/64056/21

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/Policy.swift [hunk 1]
```diff
@@ -42,6 +42,23 @@ extension Never: Identifiable {
   }
 }
 
+@available(SwiftStdlib 5.9, *)
+extension Never: Encodable {
+  @available(SwiftStdlib 5.9, *)
```
Comment (AnthonyLatsis) [inline stdlib/public/core/Policy.swift:47]: The extension’s availability should be enough.

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/Policy.swift [hunk 2]
```diff
@@ -42,6 +42,23 @@ extension Never: Identifiable {
...
+  public func encode(to: Encoder) throws {}
```
[the remark this answers is not in the document]
Comment (natecook1000) [reply stdlib/public/core/Policy.swift:48]: Good catch!

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**